### ISSUE TITLE
Dcoument conventions for command line args for tox core and plugins

### ISSUE THREAD

**Issue by obestwalter** [tox-dev/tox#498]
As @hpk42 [mentioned](https://github.com/tox-dev/tox/pull/495#issuecomment-289426993) certain conventions have been growing to prevent command line args clashes between tox core and plugins, which boil down tox -> lowercase swithces and plugins -> uppercase swtiches. This should be documented in the [plugins docs](http://tox.readthedocs.io/en/latest/plugins.html#tox.hookspecs.tox_addoption).

**Comment by gaborbernat** [comment]
I'll take this.

**Comment by gaborbernat** [comment]
I don't think this will solve things reliably. I would suggest instead plugin option should start with the plugin part, e.g. tox-travis should have --travis--a, tox-docker should have --docker-b and so on.

**Comment by gaborbernat** [comment]
We can't really enforce any of this however we can make recommendations :+1:
